### DISCUSSION TITLE
How to generate <table class="table">

### DISCUSSION THREAD

**Discussion by ferrecromain** [statiqdev/Discussions#116] [Q&A] [answered]
Hello,

I was a Wyam user, and I recently migrated to Statiq.
Before the markdown tables was generated into HTML like this: `<table class="table">` but now with Statiq it is just `<table>`

As I continue to use bootstrap, I would like to know how to customize the `table` tag.

Thanks in advance.

**Comment by TheBrambleShark** [comment]
Does kramdown format work?

```md
| A simple | table |
| with multiple | lines|
{: .my-class }
```
This would generate `<table class="my-class">...</table>`

**Reply by ferrecromain** [reply]
Thank you @Foxtrek64 for your suggestion, indeed it can work like that.

But is there a way to configure it implicitly? By editing the pipeline or something else?

**Comment by daveaglick** [comment, marked as answer]
Welcome from Wyam! This is one of the areas where Statiq (and specifically Statiq Web) is significantly more powerful than Wyam. Files are rendered based on their content type using something called [_templates_](https://www.statiq.dev/web/content-and-data/templates/). These templates can be easily modified or added without having to resort to tweaking pipelines or anything.

In your case, Statiq doesn't assume anything about CSS (we have users on bespoke, Tailwind, etc.) but this should do the trick to add the Bootstrap Markdown plugin back via the template used for Markdown files:

```csharp
using Markdig.Extensions.Bootstrap;

// ...

Bootstrapper bootstrapper = Bootstrapper.Factory
    .CreateWeb(args)
    .ModifyTemplate(
        MediaTypes.Markdown,
        x => ((RenderMarkdown)x).UseExtension<BootstrapExtension>());
```

**Reply by ferrecromain** [reply]
Yes, you are right. I appreciate the new ways to configure a website under Statiq, but sometimes I'm lost 😄.

Thank you @daveaglick for your very complete answer, I took time to understand this template mechanism and especially ```UseExtension<T>```, we can imagine developing our own extensions very easily by implementing ```IMarkdownExtension```, that's a good thing.

Using ```BootstrapExtension``` solved the problem, as we can see in the class code :
```csharp
if (node is Tables.Table)
{
    node.GetAttributes().AddClass("table");
}
```

**Reply by daveaglick** [reply]
I actually got [nerd sniped](https://xkcd.com/356/) on this a bit too and [just added support for a new `MarkdownExtensions` metadata key](https://github.com/statiqdev/statiqdev.github.io/issues/47) that can be used in the settings file or per-document in front matter:

`settings.yml`:
```yaml
MarkdownExtensions:
  - Bootstrap
```

This will go out with the next release (not sure when I'll publish it).
